eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/immutable/ImmutableArrayBag.java

```diff
@@ -246,6 +246,25 @@ public T getLast()
     }
 
     @Override
+    public T getOnly()
+    {
+        Iterator<T> iterator = this.iterator();
+
```

Comment:
If each counts of `this.counts` are always greater than `0`, my idea is following.

```java
        if (this.counts.length == 1 && this.counts[0] == 1)
        {
            return this.getFirst();
        }

        throw new IllegalStateException("Size was not 1");
```

If `this.counts` sometimes have `0`, my idea is folloing.

```java
        int size = 0;

        for (int count : this.counts)
        {
            size += count;
            if (size > 1)
            {
                throw new IllegalStateException("Size must be 1 but was greater than 1");
            }
        }

        if (size == 0)
        {
            throw new IllegalStateException("Size must be 1 but was 0");
        }

        return this.getFirst();
```